sources/models/src/modeled_types/kubernetes.rs

```diff
@@ -1324,3 +1307,98 @@ pub struct CredentialProvider {
     image_patterns: Vec<SingleLineString>,
     cache_duration: Option<KubernetesDurationValue>,
 }
+
+// =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=
+
+/// KubernetesMemoryReservationKey represents a string that contains a valid Kubernetes memory
+/// resource reservation key.
+
+#[derive(Debug, Clone, Eq, PartialEq, Hash)]
+pub struct KubernetesMemoryReservationKey {
+    inner: String,
+}
+
+#[derive(Debug, Clone, Eq, PartialEq, Hash, Deserialize)]
+enum MemoryReservationResources {
```

Comment:
Why do we need the "inner" pattern here but we were able to refactor `KubernetesEvictionHardKey` /  `EvictionSignal` into the new `KubernetesEvictionKey` enum?